scripts/develop-module.js

```diff
@@ -0,0 +1,299 @@
+const fs = require('fs');
+const path = require('path');
+const arg = require('arg');
+const log = require('./log');
+
+const DS_VERSION = 'DS6.5.v1.01';
+
+class DevelopModuleError extends Error {
+    constructor(message) {
+        super(message);
+    }
+}
+
+const args = arg({
+    '--module-name': String,
+    '--help': Boolean
+});
+const moduleName = args['--module-name'];
+const shouldHelp = args['--help'];
+
+function run() {
+    try {
+        validateArguments();
+        processRequest();
+    } catch (error) {
+        if (!(error instanceof DevelopModuleError)) {
+            throw error;
+        }
+
+        log.error(error.message);
+    }
+}
+
+function validateArguments() {
+    if (shouldHelp || moduleName) {
+        return;
+    }
+
+    throw new DevelopModuleError('Incorrect usage! Pass --help flag for help.');
+}
+
+function processRequest() {
+    if (shouldHelp) {
+        processHelpRequest();
+    } else if (moduleName) {
+        processNewModuleRequest();
+    } else {
+        throw new DevelopModuleError('Script Error! Unknown flag that passed validation to process request.');
+    }
+}
+
+function processHelpRequest() {
+    log.info(`
+USAGE:
+    --module-name
+                                The new module that you want to generate the boilerplate code for.
+                                Eg: node scripts/develop-module.js --module-name=toast-message
+    --help
+                                Used to generate this help message
+                                Eg: node scripts/develop-module.js --help
+    `);
+}
+
+function processNewModuleRequest() {
+    const newModuleData = {
```

Comment:
I see a pretty big problem here IMO, each function call you are passing this newModuleData. If you need to refactor at the future it becomes harder. I did the same thing when I wrote the other scripts and found it got too hard

I would highly recommend using a class for this. In the constructor setup the data on the class `this`. 
You can also have getters for `id` or something to keep it simple. So you can have methods like this to 
```
getImport(ds) 
   return `@import ./dist/${this.id}/${ds}/${this.id}.less`
```
without having to pass in all the newModuleData. This way you can extract a lot of the common code to keep it easier for future refactors.

Then you don't have to pass the `newModuleData` and do `newModuleData.id` everywhere. This keeps is easier to understand too.

You could initialize like this:
```
const generator = new ModuleGenerator(args)
generator.run();
```

If it's too much I'm fine with it as is, but take a look at the `SVGGenerator` in `generate-helper.js` in the scripts. It seems much cleaner to use class structures.